### PR TITLE
Fixes to horizontal zoom & tests

### PR DESCRIPTION
**Pull request by reichlfl**
Fixes the issue that zooming when already at maximum zoom level
started to move the time window.

To reproduce: Zoom in until at max zoom level. Trying to zoom further
will start moving the time window slightly to the left or right, depending
on the position of the mouse.

Also fixes some smaller inconsistencies in the tests.

### CODE REVIEW

File: src/OrbitGl/TimeGraph.cpp [hunk 1]
```diff
@@ -164,6 +164,14 @@ void TimeGraph::ZoomTime(float zoom_value, double mouse_ratio) {
   double min_time_us = ref_time_us_ - time_left / scale;
   double max_time_us = ref_time_us_ + time_right / scale;
 
+  const double duration = max_time_us - min_time_us;
+
+  if (duration < kTimeGraphMinTimeWindowsUs) {
+    const double diff = kTimeGraphMinTimeWindowsUs - duration;
+    min_time_us -= diff * mouse_ratio;
+    max_time_us += diff * (1. - mouse_ratio);
```
Comment (karupayun) [inline src/OrbitGl/TimeGraph.cpp:172]: 1.f

File: src/OrbitGl/TimeGraph.cpp [hunk 2]
```diff
@@ -187,14 +195,17 @@ void TimeGraph::VerticalZoom(float zoom_value, float mouse_world_y_pos) {
 // it.
 void TimeGraph::SetMinMax(double min_time_us, double max_time_us) {
   constexpr double kTimeGraphMinTimeWindowsUs = 0.1; /* 100 ns */
```
Comment (karupayun) [inline src/OrbitGl/TimeGraph.cpp:197]: already defined now.

**Review by beckerhe — APPROVED** [review]

**Review by karupayun — COMMENTED** [review]
As I told you offline I made it on purpose, but now that you mention it, I see that is strange for the user that something is happening although there is no zoom, so LGTM to the plan.

But I think that the solution should be 100% implemented in SetMinMax(). For example I guess that holding ctrl + right click (for selecting an interval to zoom-into) is still moving the screen although the user can't zoom-in anymore.

File: src/OrbitGl/TimeGraph.cpp (hunk 1 above)
Comment (beckerhe) [reply src/OrbitGl/TimeGraph.cpp:172]: Can you explain why? 1.f will be promoted to `double` anyway, right?

Comment (karupayun) [reply src/OrbitGl/TimeGraph.cpp:172]: sorry, you are right. We are using doubles here.

**Comment by reichlfl** [comment]
> But I think that the solution should be 100% implemented in SetMinMax(). For example I guess that holding ctrl + right click (for selecting an interval to zoom-into) is still moving the screen although the user can't zoom-in anymore.

To implement this in `SetMinMax()`, I'd need to pass in the reference timestamp that was used as "center of zooming". But this parameter is only needed when zooming with the mouse, and I didn't want to pollute the public `SetMinMax()` API (which will also be part of the `TimelineInfoInterface` later) with a confusing and often unneeded parameter. So I decided to rather implement this in `ZoomTime` because it already relates the zooming functionality to the mouse position.

As for ctrl + right click: I think the behavior makes more sense here, because it still centers the window around the selected interval even if Orbit won't zoom further.

File: src/OrbitGl/TimeGraph.cpp (hunk 2 above)
Comment (reichlfl) [reply src/OrbitGl/TimeGraph.cpp:197]: Thanks, fixed

**Pull request merged**